plots/plot.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+import argparse
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def setGlobalDefaults():
+    ## Use TrueType fonts instead of Type 3 fonts
+    #
+    # Type 3 fonts embed bitmaps and are not allowed in camera-ready submissions
+    # for many conferences. TrueType fonts look better and are accepted.
+    # This follows: https://www.conference-publishing.com/Help.php
+    matplotlib.rcParams['pdf.fonttype'] = 42
+    matplotlib.rcParams['ps.fonttype'] = 42
+
+    ## Enable tight_layout by default
+    #
+    # This ensures the plot has always sufficient space for legends, ...
+    # Without this sometimes parts of the figure would be cut off.
+    matplotlib.rcParams['figure.autolayout'] = True
+
+    ## Legend defaults
+    matplotlib.rcParams['legend.frameon'] = False
+
+
+matplotlib.rcParams['figure.figsize'] = 5, 2
+
+# Color palette
+light_gray = "#cacaca"
+dark_gray = "#827b7b"
+light_blue = "#a6cee3"
+dark_blue = "#1f78b4"
+light_green = "#b2df8a"
+dark_green = "#33a02c"
+light_red = "#fb9a99"
+dark_red = "#e31a1c"
+
+
+def save(figure, name):
+    # Do not emit a creation date, creator name, or producer. This will make the
+    # content of the pdfs we generate more deterministic.
+    metadata = {'CreationDate': None, 'Creator': None, 'Producer': None}
+
+    figure.savefig(name, metadata=metadata)
+
+    # Close figure to avoid warning about too many open figures.
+    plt.close(figure)
+
+
+# Plot an example speedup plot
+def plot_speedup():
+    labels = ['G1', 'G2', 'G3', 'G4', 'G5']
+    men_means = [1.5, 1.2, 1.3, 1.1, 1.0]
+    women_means = [1.8, 1.5, 1.1, 1.3, 0.9]
```

Comment:
Should we create another file for the data?
In the xdsl paper, we had so much data that yapf was taking 2-3s to format a file. If we had another file for the data, this would be solve.
We could see something like a file defining globals that we import directly from this file

Reply:
Yes. That would be a great idea. I am currently writing a proposal so don't have headspace to push this, but feel free to add this as another patch/PR. There are plenty such small changes that we want to do.